library/default-effects-extra.scd

```diff
@@ -120,4 +120,322 @@ SynthDef("dirt_distort" ++ ~dirt.numChannels, {|out, distort=0|
+
+	ReplaceOut.ar(out, signal)
+
+}).add;
+
+/****************************
+        SPECTRAL SCRAMBLE
+*****************************/
+
+~dirt.addModule('spectral-scram', { |dirtEvent|
+	dirtEvent.sendSynth('spectral-scram' ++ ~dirt.numChannels,
+		[
+			scram: ~scram,
+			sustain: ~sustain,
+			out: ~out
+		]
+	)
+}, { ~scram.notNil}); // play synth only if at least one of the two was given
+
+SynthDef("spectral-scram" ++ ~dirt.numChannels, { |out, scram, sustain|
+
+	var signal, chain, in, clean, teeth=256;
+
+	signal = In.ar(out, ~dirt.numChannels);
+
+    clean = signal;
+
+    chain = Array.fill(signal.size, {|i| FFT(LocalBuf(2048), signal[i])});
+
+    signal = IFFT(PV_BinScramble(chain, wipe: scram, width: scram));
+
+	ReplaceOut.ar(out, signal)
+
+}).add;
+
+/****************************
+        SPECTRAL BINSHIFT
+*****************************/
+
+~dirt.addModule('spectral-binshift', { |dirtEvent|
+	dirtEvent.sendSynth('spectral-binshift' ++ ~dirt.numChannels,
+		[
+			binshift: ~binshift,
+			sustain: ~sustain,
+			out: ~out
+		]
+	)
+}, { ~binshift.notNil}); // play synth only if at least one of the two was given
+
+SynthDef("spectral-binshift" ++ ~dirt.numChannels, { |out, binshift, sustain|
+
+	var signal, chain, in, clean, teeth=256;
+
+	signal = In.ar(out, ~dirt.numChannels);
+
+    clean = signal;
+
+    chain = Array.fill(signal.size, {|i| FFT(LocalBuf(2048), signal[i])});
+
+    signal = IFFT(PV_BinShift(chain, stretch:binshift.range(0.01,4),
+    shift:binshift.range(0,10), interp:0));
+
+	ReplaceOut.ar(out, signal)
+
+}).add;
+
+/****************************
+SPECTRAL HIGH PASS FILTER
+*****************************/
+
+~dirt.addModule('spectral-hbrick', { |dirtEvent|
+	dirtEvent.sendSynth('spectral-hbrick' ++ ~dirt.numChannels,
+		[
+			hbrick: ~hbrick,
+			sustain: ~sustain,
+			out: ~out
+		]
+	)
+}, { ~hbrick.notNil}); // play synth only if at least one of the two was given
+
+SynthDef("spectral-hbrick" ++ ~dirt.numChannels, { |out, hbrick, sustain|
+
+	var signal, chain, in, clean, teeth=256;
+
+	signal = In.ar(out, ~dirt.numChannels);
+
+    clean = signal;
+
+    chain = Array.fill(signal.size, {|i| FFT(LocalBuf(2048), signal[i])});
+
+    signal = IFFT(PV_BrickWall(chain, wipe:hbrick));
+
+	ReplaceOut.ar(out, signal)
+
+}).add;
+
+/****************************
+    SPECTRAL LOW PASS FILTER
+*****************************/
+
+~dirt.addModule('spectral-lbrick', { |dirtEvent|
+	dirtEvent.sendSynth('spectral-lbrick' ++ ~dirt.numChannels,
+		[
+			lbrick: ~lbrick,
+			sustain: ~sustain,
+			out: ~out
+		]
+	)
+}, { ~lbrick.notNil}); // play synth only if at least one of the two was given
+
+SynthDef("spectral-lbrick" ++ ~dirt.numChannels, { |out, lbrick, sustain|
+
+	var signal, chain, in, clean, teeth=256;
+
+	signal = In.ar(out, ~dirt.numChannels);
+
+    clean = signal;
+
+    chain = Array.fill(signal.size, {|i| FFT(LocalBuf(2048), signal[i])});
+
+    signal = IFFT(PV_BrickWall(chain, wipe:lbrick.range(0,-1)));
```

Comment:
`range` only applies to UGens where we know the range. Because `lbrick` is a control, it could be anything. Do you want to clip the range? Then you can use `clip(0, 1)`. You could also wrap the range `wrap(0, 1)`, then any value will be significant in some way.

Reply:
This is an old error of mine it seems. But the parameter needs to be scaled to the negative because that's when PV_BrickWall works as a low pass brick filter (positive values make it a high pass brick filter). But using range for this is wrong anyway so I'll correct this to linlin

Reply:
you could also do `.clip2.neg`, but `linlin` might be clearer.